Server/server.js

```diff
@@ -0,0 +1,73 @@
+const express = require("express");
+const app = express();
+const cors = require("cors");
+const { v4: uuidv4 } = require("uuid");
+
+app.use(express.json());
+app.use(cors());
+
+const tasks = [
+  {
+    task: "Wash dishes",
+    completed: false,
+    id: "f3a19131-91e0-4e29-86f8-528d0cb00f77",
+  },
+  {
+    task: "Do homework",
+    completed: false,
+    id: "65ed8efd-22c4-4364-99f3-23461c3e6c84",
+  },
+  {
+    task: "Cooking dinner",
+    completed: false,
+    id: "9d72980f-a815-4624-a56b-767e6545800d",
+  },
+];
+
+app.get("/tasks", function (req, res) {
+  res.send(tasks);
+});
+
+app.post("/tasks", function (req, res) {
+  const newTask = {
+    task: req.body.title,
+    completed: req.body.done || false,
+    id: uuidv4(),
+  };
+
+  tasks.push(newTask);
+  res.status(201).json(newTask);
+});
+
+app.put("/tasks/:id", function (req, res) {
+  const taskId = req.params.id;
+  console.log(`Received PUT request for task with ID ${taskId}`);
+
+  const updatedTaskIndex = tasks.findIndex((task) => task.id === taskId);

```

Comment:
you could replace Array.findIndex() with Array.find(). 
find() returns an object of task or null if not found.
Because object is not deemed to a primitive valsue, you can edit it directly, and this chansge will be done on the actual object in the array.
in this you can make the bellow look mach better: 
```js
tasks[updatedTaskIndex].title = .... ;
```